src/app/shared/nav/nav.component.html

```diff
@@ -3,7 +3,7 @@
     <nav class="navbar navbar-expand-lg navbar-dark">
       <div class="collapse navbar-collapse">
         <a class="navbar-brand" routerLink="/">Task Manager</a>
-        @if(currentUser) {
+        @if(currentUser | async) {
```

Comment:
since you have a currentUser$ observable in the component, use that instead of the behaviorSubject, otherwise there is no point to have this observable in the component.